pkg/api/persistentvolumeclaim/util_test.go

```diff
@@ -210,3 +210,139 @@ func TestDropDisabledDataSource(t *testing.T) {
 		}
 	}
 }
+
+func TestDropDisabledPVCDataSource(t *testing.T) {
+	pvcWithoutDataSource := func() *core.PersistentVolumeClaim {
+		return &core.PersistentVolumeClaim{
+			Spec: core.PersistentVolumeClaimSpec{
+				DataSource: nil,
+			},
+		}
+	}
+	apiGroup := ""
+	pvcWithDataSource := func() *core.PersistentVolumeClaim {
+		return &core.PersistentVolumeClaim{
+			Spec: core.PersistentVolumeClaimSpec{
+				DataSource: &core.TypedLocalObjectReference{
+					APIGroup: &apiGroup,
+					Kind:     "PersistentVolumeClaim",
+					Name:     "test_clone",
+				},
+			},
+		}
+	}
+
+	pvcInfo := []struct {
+		description   string
+		hasDataSource bool
+		pvc           func() *core.PersistentVolumeClaim
+	}{
+		{
+			description:   "pvc without DataSource",
+			hasDataSource: false,
+			pvc:           pvcWithoutDataSource,
+		},
+		{
+			description:   "pvc with DataSource",
+			hasDataSource: true,
+			pvc:           pvcWithDataSource,
+		},
+		{
+			description:   "is nil",
+			hasDataSource: false,
+			pvc:           func() *core.PersistentVolumeClaim { return nil },
+		},
+	}
+
+	for _, enabled := range []bool{true, false} {
+		for _, oldpvcInfo := range pvcInfo {
```

Comment:
I think this variable name maybe misleading and I am not sure If it is necessary to loop the pvcInfo for twice